packages/loader/container-loader/src/container.ts

```diff
@@ -225,7 +225,19 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     private storageService: IDocumentStorageService | undefined | null;
     private blobsCacheStorageService: IDocumentStorageService | undefined;
 
+    /**
+     * The current client id
+     */
     private _clientId: string | undefined;
+    /**
+     * Our current (and previous) clientIds, with the latest (and current) at the
+     * end of the array.  We keep a history so that messages with different IDs
+     * can be identified as originating from this client in cases of multiple
+     * disconnects/reconnects.  _clientId exists above as well to minimize/stage
+     * this change
+     */
+    private readonly clientIdHistory: string[] = [];
+    private readonly maxClientIdHistory = 5;
```

Comment:
IMO don't worry about any max here.

Reply:
we can increase it but i'd be wary of unlimiting it because of long-running sessions/network instability

Reply:
Why is that?

Reply:
2 problems. not clearing it is a possible memory leak. we also search this on every ops, so it could impact per if i gets large. i think making it like 100 would still be safe